### DISCUSSION TITLE
How to update a dependent version of a private package , but not publish it, when use the publish command

### DISCUSSION THREAD

**Discussion by greper** [lerna/lerna#3533] [Q&A]
I know I can use lerna publish --no-private to not publish private packages,  But in this way ,dependencies in the private package are not automatically updated

**Comment by fahslaj** [comment]
Hi @greper! By "private package", I assume you mean with the property "private": true in its package.json. In that case, Lerna will automatically skip over private packages when publishing to the remote registry. You should be able to just run `lerna publish` without `--no-private`, which will update the versions of all packages, including private ones, but only publish the non-private ones to the remote registry.

**Reply by greper** [reply]
Oh, I see, my private package is the submodule, and although it won't be published, it doesn't even pass the version command

**Reply by greper** [reply]
when i use lerna publish without --no-private

**Reply by greper** [reply]
Thanks, after I made the following changes to `node_modules/@lerna/version/lib/git-add.js` , all the annoying problems about submodule are gone

The ` git add . ` Command can do the same thing with no more errors


```

  log.silly("gitAdd", files);

  // return childProcess.exec("git", ["add", "--", ...files], execOpts);
  //                                    ↓↓↓↓↓↓↓
  return childProcess.exec("git", ["add", "."], execOpts);
}

```

**Reply by greper** [reply]
This might make it an option for those who need it
